modules/repository/repo.go

```diff
@@ -70,6 +72,75 @@ func MigrateRepositoryGitData(ctx context.Context, u *models.User, repo *models.
 		return repo, fmt.Errorf("Clone: %v", err)
 	}
 
+	if opts.LFS {
+		_, err = git.NewCommand("lfs", "fetch", opts.CloneAddr).RunInDir(repoPath)
```

Comment:
This PR needs more comments to explain what it's doing.

This is primarily because not all of its know what git LFS fetch actually does.

Git LFS fetch here is downloading all of the assets directly to the git repo store and then you are walking the assets to copy them in to LFS asset store.

Of course - this makes at least three deadly assumptions that are going to break.

1. LFS files are not always stored on a physical filesystem
2. The LFS files may not be even temporarily wanted to go in to git repository store - large files could end up being mirrored before you move them.
3. Git LFS fetch does not obey max file constraints for repos.

There are several other problems with git LFS fetch that would need sorting in a addition too.

Reply:
`git-lfs` env LfsStorageDir can be set to LFS_CONTENT_PATH/tmp and they will be downloaded directly to the gitea storage.
As for the file max size constraints: can you please specify where I can find them? The files exceeding the size may be excluded.

Reply:
Ok now I made it to fetch files to gitea lfs store.
Can you please specify where I can set/get max file constraints so I can exclude files exceeding limit from downloading?

Reply:
The LFS storage isn't necessarily even on a filesystem, and then even if it was, because of the difference in the format of the LFS filesystem you're then resharding it whilst live within the LFS storage for all of the repos and you're not accounting for collisions where the SHA256 is OID is already in the db.

Further you've not handled any of the max file constraints here. 

I really don't think simply calling `git lfs fetch` is going to work.

Reply:
Git LFS smudge or git LFS filter-process might be closer to be being able to be redirected to where we want things to go - however they'd need to be checked to see if they would dump stuff in the LFS dB.

Git LFS ls-files --all could be used to list all the pointer files in the git object db.

Reply:
Added max file constraints.
What it means for LFS storage to be not present on a filesystem, is it can be set to URL?

Reply:
It can be stored on minio for example